GDJS/Runtime/runtimeobject.ts

```diff
@@ -2323,26 +2332,32 @@ namespace gdjs {
       }
 
       // Do a real check if necessary.
-      let testSqDist = closest ? raySqBoundingRadius : 0;
-      const hitBoxes = this.getHitBoxesAround(x, y, endX, endY);
-      for (const hitBox of hitBoxes) {
-        const res = gdjs.Polygon.raycastTest(hitBox, x, y, endX, endY);
-        if (res.collision) {
-          if (closest && res.closeSqDist < testSqDist) {
-            testSqDist = res.closeSqDist;
-            result = res;
-          } else {
-            if (
-              !closest &&
-              res.farSqDist > testSqDist &&
-              res.farSqDist <= raySqBoundingRadius
-            ) {
-              testSqDist = res.farSqDist;
-              result = res;
-            }
+      if (closest) {
+        let sqDistMin = raySqBoundingRadius;
+        const hitBoxes = this.getHitBoxesAround(x, y, endX, endY);
+        for (const hitBox of hitBoxes) {
+          const res = gdjs.Polygon.raycastTest(hitBox, x, y, endX, endY);
+          if (res.collision && res.closeSqDist < sqDistMin) {
+            sqDistMin = res.closeSqDist;
+            gdjs.Polygon.copyRaycastTestResult(res, result);
+          }
+        }
+      } else {
+        let sqDistMax = 0;
+        const hitBoxes = this.getHitBoxesAround(x, y, endX, endY);
+        for (const hitBox of hitBoxes) {
+          const res = gdjs.Polygon.raycastTest(hitBox, x, y, endX, endY);
+          if (
+            res.collision &&
+            res.farSqDist > sqDistMax &&
```

Comment:
sqDistMax should probably be set `-Number.MAX_VALUE` to allow a point at a distance = 0 to be detected. I guess `raycastTest` already fulfill the role to ensure values under 0 won't be returned.

The same thing could be done for `sqDistMax`.

Reply:
> The same thing could be done for sqDistMax

You mean sqDistMin?

Reply:
yes 😄

Reply:
I did the change but I'm not sure there is anything to do for sqDistMax, it's already initialised at the maximum radius we are allowing the ray to go.

Reply:
Do you mean sqDistMin?